language/printer/printer.go

```diff
@@ -372,9 +372,22 @@ var printDocASTReducer = map[string]visitor.VisitFunc{
 	"StringValue": func(p visitor.VisitFuncParams) (string, interface{}) {
 		switch node := p.Node.(type) {
 		case *ast.StringValue:
-			return visitor.ActionUpdate, `"` + fmt.Sprintf("%v", node.Value) + `"`
-		case map[string]interface{}:
-			return visitor.ActionUpdate, `"` + getMapValueString(node, "Value") + `"`
+			str := node.Value
+			// Handle block-strings
+			// Section 2.9.4 graphql spec block string is
+			// identical to standard quoted string with
+			// newlines escaped.
+			if strings.Contains(str, "\n") {
+				str = strings.ReplaceAll(str, "\n", "\\n")
```

Comment:
Two questions:
- are there any other quote cases [we can figure that out later]
- are you better off using the triple quote block-string in the case you are quoting the \n?